extensions/wsc/src/main/java/org/mule/extension/ws/api/security/WssVerifySignatureSecurityStrategy.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.ws.api.security;
+
+
+import static org.apache.ws.security.handler.WSHandlerConstants.SIGNATURE;
+import static org.apache.ws.security.handler.WSHandlerConstants.SIG_PROP_REF_ID;
+import static org.mule.extension.ws.internal.security.SecurityStrategyType.INCOMING;
+import org.mule.extension.ws.internal.security.EncryptionHelper;
+import org.mule.extension.ws.internal.security.SecurityStrategyType;
+import org.mule.extension.ws.internal.security.callback.WSPasswordCallbackHandler;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.tls.TlsContextFactory;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+
+/**
+ * Verifies the signature of a SOAP response, using certificates of the trust-store in the provided TLS context.
+ *
+ * @since 4.0
+ */
+public class WssVerifySignatureSecurityStrategy implements SecurityStrategy {
```

Comment:
should we enable the reuse of this strategies by allowing `topLevelElement` declarations?